text/00110-compatibility-strategy.md

```diff
@@ -0,0 +1,364 @@
+---
+feature name: compatibility-strategy
+start date: February 2nd, 2020
+rfc pr: https://github.com/aws/aws-cdk-rfcs/pull/111
+related issue: https://github.com/aws/aws-cdk-rfcs/issues/110
+---
+
+<!-- replace the blockquoted sections with your content -->
+# Summary
+
+There are currently two statements we advertise to our customers with regards to framework and cli compatibility:
+
+- CLI versions should always support older framework versions. That is, upgrading the CLI should never require an upgrade of the framework.
+- Framework versions might require an upgrade of the CLI. In which case, we provide instructions as to which CLI version should be used.
+
+This RFC details the necessary strategy and mechanism for ensuring that we keep those promises when releasing new versions.
+
+# Motivation
+
+There are 2 main motivations why we should put some substantial effort into this:
+
+## (1) Haven't we suffered enough?
+
+Recently, we experienced several incidents that are directly related to this domain:
+
+- https://github.com/aws/aws-cdk/issues/5986
+- https://github.com/aws/aws-cdk/issues/4897
+- https://github.com/aws/aws-cdk/issues/4294
+
+These issues were very awkward and caused both our customers and ourselves a lot of unwanted work.
+They also reflect poorly on the stability of the CDK, especially in a post GA era.
+
+*We want to implement this RFC to prevent (or at least greatly reduce) the occurrences of such issues.*
+
+## (2) Refactor CX Protocol versioning
+
+We've talked about the mechanism we currently use to perform protocol version changes quite a bit. The process and the expected output aren't very clear, and we seem to be always trying to understand it again and again.
+This might warrant a re-think/re-design, to make it much more ergonomic and developer friendly. 
+
+However, performing changes on such a crucial part of the system, shouldn't be done without a proper validation mechanism.
+
+*We want to implement this RFC so that we are able to iterate and change the versioning mechanism itself* 
+
+Actually, We have already acknowledged the need for this a while back:
+
+- https://github.com/aws/cdk-ops/issues/365
+- https://github.com/aws/cdk-ops/issues/233
+
+Its time we solve this once and for all.
+
+After implementing this RFC, we can expect the following outcomes:
+
+- An incompatible change in the CLI won't make it to the release.
+- An incompatible change in the framework, will always be accompanied with a version bump.
```

Comment:
What do you mean by version bump? Protocol version?

Reply:
Yes